llvm/lib/Target/AMDGPU/AMDGPUAsmPrinter.cpp

```diff
@@ -420,38 +421,41 @@ uint16_t AMDGPUAsmPrinter::getAmdhsaKernelCodeProperties(
   return KernelCodeProperties;
 }
 
-amdhsa::kernel_descriptor_t AMDGPUAsmPrinter::getAmdhsaKernelDescriptor(
-    const MachineFunction &MF,
-    const SIProgramInfo &PI) const {
+MCKernelDescriptor
+AMDGPUAsmPrinter::getAmdhsaKernelDescriptor(const MachineFunction &MF,
+                                            const SIProgramInfo &PI) const {
   const GCNSubtarget &STM = MF.getSubtarget<GCNSubtarget>();
   const Function &F = MF.getFunction();
   const SIMachineFunctionInfo *Info = MF.getInfo<SIMachineFunctionInfo>();
 
-  amdhsa::kernel_descriptor_t KernelDescriptor;
-  memset(&KernelDescriptor, 0x0, sizeof(KernelDescriptor));
+  auto CreateExpr = [&MF](int64_t Value) {
+    return MCConstantExpr::create(Value, MF.getContext());
+  };
```

Comment:
This lambda isn't doing much useful, just directly call the MCConstantExpr::create?

Reply:
I suggested to use a lambda to reduce the noise, otherwise you end up with very verbose lines that have a ton of duplicate information (`MCConstantExpr::create(X, MF.getContext())` where only `X` changes)

Reply:
It's less noisy if you pull the MF.getContext() into a variable